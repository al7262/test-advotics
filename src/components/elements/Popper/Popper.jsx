/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable react/forbid-prop-types */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

export default function Popper(props) {
  const {
    children, className, onClose, open, popperRef, style,
  } = props;
  const [content, setContent] = useState(null);
  const customClass = [
    styles.root,
    !content && styles['popper-closed'],
    className,
  ].filter(Boolean).join(' ');

  const popperClose = (event) => {
    if (popperRef && !popperRef.contains(event.target)) {
      onClose();
      setContent(null);
    }
  };

  useEffect(() => {
    if (open) {
      setContent(children);
      window.addEventListener('click', popperClose);
    }

    return () => {
      setContent(null);
      window.removeEventListener('click', popperClose);
    };
  }, [open, children]);

  return (
    <section className={customClass} onClick={(e) => e.stopPropagation()} style={style}>
      {content}
    </section>
  );
}

Popper.defaultProps = {
  children: null,
  className: '',
  onClose: () => { },
  open: false,
  popperRef: null,
  style: {},
};

Popper.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClose: PropTypes.func,
  open: PropTypes.bool,
  popperRef: PropTypes.object,
  style: PropTypes.object,
};
