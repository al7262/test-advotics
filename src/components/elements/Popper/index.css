.root {
  background-color: var(--white-color);
  border-radius: 0.25rem;
  filter: var(--drop-shadow-l);
  position: absolute;
  left: 0;
  top: 0;
  z-index: 10;
}

.popper-closed {
  display: none !important;
}