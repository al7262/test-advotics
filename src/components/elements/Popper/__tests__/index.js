/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Popper from '../Popper';

describe('src/components/elements/Popper', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Popper />);
    expect(tree).toMatchSnapshot();
  });
});
