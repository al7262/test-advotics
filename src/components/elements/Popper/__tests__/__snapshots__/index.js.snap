// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Popper render 1`] = `
<section
  className=""
  onClick={[Function]}
  style={Object {}}
/>
`;
